src/Microsoft.AspNetCore.Http.Abstractions/Extensions/HeaderDictionaryExtensions.cs

```diff
@@ -11,6 +11,7 @@ public static class HeaderDictionaryExtensions
         /// <summary>
         /// Add new values. Each item remains a separate array entry.
         /// </summary>
+        /// <param name="headers">The <see cref="IHeaderDictionary"/> that this method extends.</param>
```

Comment:
It's named Append because it will append to existing values for the same key (if any). You're not just appending to the dictionary.